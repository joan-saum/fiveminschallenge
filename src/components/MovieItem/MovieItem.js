// @flow

import React, {useCallback, useMemo} from 'react';
import {View, Text, Image, TouchableWithoutFeedback} from 'react-native';
import type {Node} from 'react';

import styles from './MovieItem.style';

type Props = {|
  onPress: (id: number) => void,
  image: string,
  title: string,
  overview: string,
  id: number,
|};

const MovieItem = ({image, title, overview, id, onPress}: Props): Node => {
  const onPressItem = useCallback(() => {
    onPress(id);
  }, [onPress, id]);

  const source = useMemo(
    () => ({
      uri: image,
    }),
    [image],
  );

  return (
    <TouchableWithoutFeedback onPress={onPressItem}>
      <View style={styles.container}>
        <View style={styles.imageContainer}>
          <Image source={source} style={styles.image} />
        </View>
        <View style={styles.content}>
          <Text numberOfLines={1} style={styles.title}>
            {title}
          </Text>
          <Text
            numberOfLines={4}
            textBreakStrategy="simple"
            style={styles.text}>
            {overview}
          </Text>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default MovieItem;
