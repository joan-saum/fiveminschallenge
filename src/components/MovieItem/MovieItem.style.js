import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    marginLeft: 20,
    marginRight: 20,
    height: 140,
    marginBottom: 20,
    borderRadius: 10,
    borderWidth: 1,
    overflow: 'hidden',
  },
  imageContainer: {
    display: 'flex',
    width: 120,
  },
  image: {
    flex: 1,
  },
  content: {
    padding: 10,
    flexShrink: 1,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 5,
  },
  text: {
    fontSize: 14,
  },
});

export default style;
