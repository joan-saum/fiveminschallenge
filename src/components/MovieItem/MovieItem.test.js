// @flow

import * as React from 'react';
import renderer from 'react-test-renderer';
import {TouchableWithoutFeedback} from 'react-native';

import MovieItem from './MovieItem';

jest.unmock('./MovieItem');

const defaultProps = {
  image: 'image',
  title: 'title',
  overview: 'overview',
  id: 1,
  onPress: jest.fn().mockName('onPress'),
};

describe('MovieItem', () => {
  it('should render a MovieItem', () => {
    const tree = renderer.create(<MovieItem {...defaultProps} />);

    expect(tree).toMatchSnapshot();
  });

  it('should call onPress when clicking on the item', () => {
    const onPress = jest.fn().mockName('onPress');
    const tree = renderer.create(
      <MovieItem {...defaultProps} onPress={onPress} />,
    );

    expect(onPress).toHaveBeenCalledTimes(0);

    tree.root.findByType(TouchableWithoutFeedback).props.onPress();

    expect(onPress).toHaveBeenCalledTimes(1);
    expect(onPress).toHaveBeenCalledWith(1);
  });
});
