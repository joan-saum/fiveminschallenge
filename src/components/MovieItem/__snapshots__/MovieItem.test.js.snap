// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem should render a MovieItem 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 10,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 140,
      "marginBottom": 20,
      "marginLeft": 20,
      "marginRight": 20,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "width": 120,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "image",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexShrink": 1,
        "padding": 10,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      title
    </Text>
    <Text
      numberOfLines={4}
      style={
        Object {
          "fontSize": 14,
        }
      }
      textBreakStrategy="simple"
    >
      overview
    </Text>
  </View>
</View>
`;
